core/trino-main/src/main/java/io/trino/operator/scalar/ArrayReduceFunction.java

```diff
@@ -60,7 +60,6 @@ private ArrayReduceFunction()
                         .build())
                 .nullable()
                 .argumentNullability(false, true, false, false)
-                .nondeterministic()
```

Comment:
> Non-determinism depends on the body of the lambda expressions
passed to the function.
> That's handled elsewhere.

Where can I find that?

Reply:
It’s handled by DeterminismEvaluator.
It’s may not be immediately obvious that lambdas are handled, but that class walks the expression tree and returns “not deterministic” if any call contained in the expression is non-deterministic.

Reply:
yup, I see it now. It's a surprisingly simple implementation too (of the DeterminismEvaluator(s)).
Looks good to merge IMO.